modules/custom-status/meta/required-fields-cron-cleaner.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * class ReqMetadataFieldsCronCleaner
+ * Cleans up fields that are required on a custom status, and should be cleaned up.
+ */
+
+namespace VIPWorkflow\Modules\CustomStatus\Meta;
+
+use VIPWorkflow\Modules\Custom_Status;
+use VIPWorkflow\VIP_Workflow;
+
+class RequiredFieldsCronCleaner {
+
+	public static function init(): void {
+		// Remove deleted users from required users, and if a user is reassigned, update the required users
+		add_action( 'delete_user', [ __CLASS__, 'remove_deleted_user_from_required_users' ], 10, 2 );
```

Comment:
I think this is a step toward more complexity from the prior implementation in `required-user-ids.php`, because we're combining the cleanup of several different types of metadata into a larger file, as opposed to having small encapsulated behaviors in meta-specific files. I see what you're trying to do here with combining functionality, but overall I think [the prior logic](https://github.com/Automattic/vip-workflow-plugin/blob/643ec524308f8103af9bafbc5618164d31398b92/modules/custom-status/meta/required-user-ids.php#L14-L41) is easy to read, simple, and directly scoped to the metadata type. These two filters aren't really connected other than the fact that they have similar logic for filtering a list.

If we add cleanup for new features, we'll probably need to expand this file to support those types, or if it doesn't fit this abstraction we'll need to change the abstraction or expand the file with unrelated code. Where possible, I think the best way to fight complexity is to keep file sizes smaller and class responsibilities self-contained, and I'm feeling like the design here goes in the opposite direction.

Reply:
Talked it over, and we will do a hybrid approach:

- Keep the individual files focused on cleaning up one field
- Move the common code to a utility file